src/Files.App/UserControls/TabBar/TabBar.xaml.cs

```diff
@@ -366,5 +374,25 @@ private void TabViewItem_Loaded(object sender, RoutedEventArgs e)
 				});
 			}
 		}
+
+		private void DragAreaRectangle_Loaded(object sender, RoutedEventArgs e)
+		{
+			if (HorizontalTabView.ActualWidth > 0 && TabBarAddNewTabButton.Width > 0 && TitleBarWidth.Value > 0)
+			{
+				HorizontalTabView.Measure(new(HorizontalTabView.ActualWidth - TabBarAddNewTabButton.Width - TitleBarWidth.Value, HorizontalTabView.ActualHeight));
+				return;
+			}
+
+			DispatcherQueue.TryEnqueue(() => DragAreaRectangle_Loaded(sender, e));
+		}
+
+		public int SetTitleBarDragRegion(InputNonClientPointerSource source, SizeInt32 size, double scaleFactor, Func<UIElement, RectInt32?, RectInt32> getScaledRect)
```

Comment:
Is it possible to move this method back to `MainPage`?

Reply:
@0x5bfa  suggested to convert all TabBar operations into one file, and for me it is reasonable

Reply:
I don't think we want to restrict the logic to the TabBar, this way all our options are open if we ever add a scenario where the TabBar is hidden.

Reply:
I don't know, my preference is to keep the changes

Reply:
I would prefer if this is reverted